client/verta/verta/operations/monitoring/alert/status/_status.py

```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+
+import abc
+
+from .....external import six
+
+from ....._protos.public.monitoring import Alert_pb2 as _AlertService
+
+
+# TODO: move into separate files
+@six.add_metaclass(abc.ABCMeta)
+class _AlertStatus(object):
+    _ALERT_STATUS = _AlertService.AlertStatusEnum.UNKNOWN
+
+    def __repr__(self):
+        return "<{} alert status>".format(
+            _AlertService.AlertStatusEnum.AlertStatus.Name(self._ALERT_STATUS).lower()
+        )
+
+    @classmethod
+    def _from_proto(cls, msg):
+        """
+        Returns an alert status object.
+
+        Parameters
+        ----------
+        msg : int
+            Variant of ``AlertStatusEnum``.
+
+        Returns
+        -------
+        :class:`_AlertStatus` subclass
+
+        """
+        SUBCLASSES = [
```

Comment:
I've been conflicted about using `__subclasses__` because based on what I've read, it seems to be an internal detail of the CPython implementation. Have you seen it used reliably elsewhere, out of curiousity?

Reply:
Interesting. Nothing I read seemed to indicate that it was a CPython thing. It's documented in the reference after all. I'm fine with taking this approach if we're in doubt / until we know vet that it's not an implementation specific thing.

Reply:
For completeness:
my
> seems to be an internal detail of the CPython implementation

perception comes from [this frequently cited old entry in the Python mailing list](https://mail.python.org/pipermail/python-list/2003-August/210297.html):
> In short, \_\_subclasses__ exists to support efficient runtime class mutation
at the C level.

and the [Python reference](https://docs.python.org/2/library/stdtypes.html?highlight=__subclasses__#special-attributes) mentions (emphasis mine):
> **The implementation** adds a few special read-only attributes to several object types . . .

If I find some time to get PyPy running on my machine, I could verify this eventually 😅